tracer/src/Datadog.FleetInstaller/GlobalEnvVariableHelper.cs

```diff
@@ -0,0 +1,89 @@
+// <copyright file="GlobalEnvVariableHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+
+namespace Datadog.FleetInstaller;
+
+internal static class GlobalEnvVariableHelper
+{
+    public static bool SetMachineEnvironmentVariables(
+        ILogger log,
+        TracerValues values,
+        out Dictionary<string, string> previousValues)
+    {
+        previousValues = new();
+        try
+        {
+            log.WriteInfo("Installing global environment variables");
+
+            foreach (var kvp in values.GlobalRequiredEnvVariables)
+            {
+                var previousValue = Environment.GetEnvironmentVariable(kvp.Key, EnvironmentVariableTarget.Machine);
+                if (!string.IsNullOrWhiteSpace(previousValue))
+                {
+                    previousValues[kvp.Key] = previousValue;
```

Comment:
What happens when there is no previous value and we set it, and then we revert it?

Since the revert is just looking at the previous would the new values not be reverted? 
Unsure if that case can even really happen

Reply:
Maybe they get removed in `RemoveMachineEnvironmentVariables`

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/7110/commits/546d38fa10212f962f58c2866247f8ee86a7a43a